File: src/variable_manager.cpp
#include "variable_manager.h"
#include <unordered_map>

VariableManager::VariableManager() = default;

VariableManager& VariableManager::set(const std::string& name, std::string_view value) {
    variables[name] = std::string(value);
    return *this;
};

VariableManager& VariableManager::set_exported(const std::string& name, std::string_view value) {
    variables[name] = std::string(value);
    exported.insert(name);
    return *this;
}

VariableManager& VariableManager::mark_exported(const std::string& name) {
    variables.try_emplace(name, "");
    exported.insert(name);
    return *this;
}

std::string_view VariableManager::get(const std::string& name) {
    if (auto it = variables.find(name); it != variables.end()) {
        return it->second;
    }
    return "";
}

bool VariableManager::exist(const std::string& name) {
    return variables.find(name) != variables.end();
}

std::vector<std::string> VariableManager::environment() const {
    std::vector<std::string> env;
    env.reserve(exported.size());
    for (const auto& name : exported) {
        if (auto it = variables.find(name); it != variables.end()) {
            env.push_back(name + "=" + it->second);
        }
    }
    return env;
}

File: src/variable_manager.rs
// variable_manager.rs

use std::collections::HashMap;

#[derive(Debug, Default)]
pub struct VariableManager {
    variables: HashMap<String, String>,
}

impl VariableManager {
    pub fn new() -> Self {
        VariableManager {
            variables: HashMap::new(),
        }
    }

    pub fn set(&mut self, name: &str, value: &str) -> &mut Self {
        self.variables.insert(name.to_string(), value.to_string());
        self
    }

    pub fn get(&self, name: &str) -> Option<&String> {
        self.variables.get(name)
    }

    pub fn exist(&self, name: &str) -> bool {
        self.variables.contains_key(name)
    }
}
